tests/core/models/test_edition.py

```diff
@@ -332,12 +332,48 @@ def test_calculate_presentation_very_long_author(
         for i in range(0, 500):
             author, ignore = db.contributor(sort_name=f"AuthorLast{i}, AuthorFirst{i}")
             authors.append(author.sort_name)
+
         wr = db.edition(authors=authors)
         author, sort_author = wr.calculate_author()
         wr.calculate_presentation()
         db.session.commit()
-        assert len(wr.author) == 1024
-        assert len(wr.sort_author) == 1024
+
+        def do_check(original_str: str, truncated_str: str):
+            assert (
+                len(truncated_str)
+                == Edition.SAFE_AUTHOR_FIELD_LENGTH_TO_AVOID_PG_INDEX_ERROR
+            )
+            assert truncated_str.endswith("...")
+            assert (
+                len(original_str)
+                > Edition.SAFE_AUTHOR_FIELD_LENGTH_TO_AVOID_PG_INDEX_ERROR
+            )
+            assert not original_str.endswith("...")
+
+        do_check(author, wr.author)
+        do_check(sort_author, wr.sort_author)
+
+    def test_calculate_presentation_shortish_author(
+        self, db: DatabaseTransactionFixture
+    ):
+        authors = []
+        author, ignore = db.contributor(sort_name=f"AuthorLast{i}, AuthorFirst{i}")
+        authors.append(author.sort_name)
+        wr = db.edition(authors=authors)
+        author, sort_author = wr.calculate_author()
+        wr.calculate_presentation()
+        db.session.commit()
+
+        def do_check(original_str: str, truncated_str: str):
+            assert truncated_str == original_str
+            assert not truncated_str.endswith("...")
+            assert (
+                len(original_str)
+                < Edition.SAFE_AUTHOR_FIELD_LENGTH_TO_AVOID_PG_INDEX_ERROR
```

Comment:
Should this be `<=` on l.372?

Probably good to also ensure that the original does not already end in ellipses, as we did above:
```
    assert not original_str.endswith("...")
```

Reply:
Fixed.